### PR TITLE
adding avatars functionality

### PR DESCRIPTION
**Pull request by MaxaKozin**

### CODE REVIEW

File: api/users/user/user.controller.js [hunk 1]
```diff
@@ -41,8 +43,25 @@ const currentUserController = async (req, res, next) => {
   }
 };
 
+const uploadUserAvatarAndUpdate = async (req, res, next) => {
+  try {
+    const file = req.file;
+    if (file.filename) {
+      const avatarPath = `http://localhost:3000/images/${file.filename}`;
```
Comment (odykyi) [inline api/users/user/user.controller.js]: move host and port to config

File: middlewares/upload.middleware.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const multer = require("multer");
+
+const avatarLoader = () => {
+  const storage = multer.diskStorage({
+    destination: (req, file, cb) => {
+      cb(null, "public/images");
```
Comment (odykyi) [inline middlewares/upload.middleware.js]: move "public/images" to config

File: services/avatar.service.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+const generate = async (id) => {
+  const avatar = new AvatarGenerator({
+    parts: ["background", "face", "clothes", "head", "hair", "eye", "mouth"],
+    partsLocation: path.join(__dirname, "../img"),
+    imageExtension: ".png",
+  });
+  const variant = "male";
+  const image = await avatar.generate("email@example.com", variant);
+  image.png().toFile(`tmp/${id}.png`);
```
Comment (odykyi) [inline services/avatar.service.js]: move `tmp/${id}.png` to config

export const youFunctionName = (id ) => `tmp/${id}.png`

File: services/avatar.service.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const updateUserAvatarUrl = async (id) => {
+  const avatarUrl = `http://localhost:3000/images/${id}.png`;
```
Comment (odykyi) [inline services/avatar.service.js]: move host and port to config

File: services/avatar.service.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const oldFilePath = path.resolve("tmp", `${id}.png`);
+  const newFilePath = path.resolve("public", "images", `${id}.png`);
+  await fs.move(oldFilePath, newFilePath, (e) => {
+    if (e) return console.log(e);
+  });
+};
+
+const deleteDefaultAvatar = async (id) => {
+  const defaultAvatarPath = path.resolve("public", "images", `${id}.png`);
+  await fs
```
Comment (odykyi) [inline services/avatar.service.js]:
```suggestion
  try {
    await fs.readFile(defaultAvatarPath)
    await fs.unlink(defaultAvatarPath);
 }
    catch (e ) {
      // handle error
    }
```

File: services/avatar.service.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+const updateUserAvatarUrl = async (id) => {
+  const avatarUrl = `http://localhost:3000/images/${id}.png`;
+
+  return await User.updateUser(id, { avatarURL: avatarUrl });
+};
+
+const moveAvatar = async (id) => {
+  const oldFilePath = path.resolve("tmp", `${id}.png`);
+  const newFilePath = path.resolve("public", "images", `${id}.png`);
+  await fs.move(oldFilePath, newFilePath, (e) => {
```
Comment (odykyi) [inline services/avatar.service.js]: wrap to try catch and remove callback 

 await fs.move(oldFilePath, newFilePath)

File: services/avatar.service.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+const updateUserAvatarUrl = async (id) => {
+  const avatarUrl = `http://localhost:3000/images/${id}.png`;
+
+  return await User.updateUser(id, { avatarURL: avatarUrl });
+};
+
+const moveAvatar = async (id) => {
+  const oldFilePath = path.resolve("tmp", `${id}.png`);
+  const newFilePath = path.resolve("public", "images", `${id}.png`);
```
Comment (odykyi) [inline services/avatar.service.js]: move newFilePath to config

File: services/avatar.service.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const updateUserAvatarUrl = async (id) => {
+  const avatarUrl = `http://localhost:3000/images/${id}.png`;
+
+  return await User.updateUser(id, { avatarURL: avatarUrl });
+};
+
+const moveAvatar = async (id) => {
+  const oldFilePath = path.resolve("tmp", `${id}.png`);
```
Comment (odykyi) [inline services/avatar.service.js]: move oldFilePath to config

File: services/avatar.service.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+const AvatarGenerator = require("avatar-generator");
+const path = require("path");
+const fs = require("fs-extra");
+const User = require("../api/users/user/user.model");
+
+const generate = async (id) => {
+  const avatar = new AvatarGenerator({
+    parts: ["background", "face", "clothes", "head", "hair", "eye", "mouth"],
+    partsLocation: path.join(__dirname, "../img"),
```
Comment (odykyi) [inline services/avatar.service.js]: move partsLocation to config

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

File: api/users/user/user.controller.js (hunk 1 above)
Comment (MaxaKozin) [reply api/users/user/user.controller.js]: moved host and port to dotenv

File: middlewares/upload.middleware.js (hunk 2 above)
Comment (MaxaKozin) [reply middlewares/upload.middleware.js]: moved to dotenv

File: services/avatar.service.js (hunk 3 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: moved to created config service

File: services/avatar.service.js (hunk 4 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: moved to dotenv

File: services/avatar.service.js (hunk 5 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: refactored according suggested change

File: services/avatar.service.js (hunk 6 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: wrapped into try catch

File: services/avatar.service.js (hunk 7 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: created function in config service

File: services/avatar.service.js (hunk 8 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: created function in config service

File: services/avatar.service.js (hunk 9 above)
Comment (MaxaKozin) [reply services/avatar.service.js]: created function in config service

**Review by odykyi — APPROVED** [review]

**Pull request merged**